Assignment6.java

```diff
@@ -31,36 +34,87 @@
             // If i isn't already empty.
             if (indexMap.get(0) != i) {
                 // Move the number in i to the empty slot so that i becomes empty.
-                Move emptyCurrentSlot = new Move(given[i], i, indexMap.get(0));
-                
-                listOfMoves.add(emptyCurrentSlot);
-                move(emptyCurrentSlot, given, indexMap);
+                listOfMoves.add(move(given[i], given, indexMap));

```

Comment:
I have the impression we don't need indexMap. Instead, we could simply compare given[i] with desired[i].